File: 1331-rank-transform-of-an-array/1331-rank-transform-of-an-array.cpp
class Solution {
public:
    vector<int> arrayRankTransform(vector<int>& arr) {
        set<int>st;
        
        for(int i=0;i<arr.size();i++){
            st.insert(arr[i]);
        }

        unordered_map<int,int>mp;
        int rk=1;
        
        for(int num:st){
            mp[num]=rk;
            rk++;
        }

        for(int i=0;i<arr.size();i++){
            arr[i]=mp[arr[i]];
        }

        return arr;

    }
};

File: 1331-rank-transform-of-an-array/1331-rank-transform-of-an-array.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        new_arr = sorted(set(arr))
        d = {}
        l = []
        for i,j in enumerate(new_arr):
            d[j] = i+1
        for i in arr:
            l.append(d[i])     
        return l   